zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -291,7 +314,10 @@ public Interpreter getCurrentRepl() {
 
     String replName = getRequiredReplName(buffer);
     if (replName != null && cursor > replName.length()) {
-      cursor -= replName.length() + 1;
+      Matcher m = Pattern.compile(String.format("%%%s\\s*", replName)).matcher(buffer);
```

Comment:
I'm not sure we should have another way to match here? should we correct/reuse the logic in L307, or getRequiredReplName etc?